ports/mimxrt/machine_adc.c

```diff
@@ -104,6 +105,11 @@ static mp_int_t mp_machine_adc_read_u16(machine_adc_obj_t *self) {
     LPADC_GetDefaultConvCommandConfig(&adc_config);
     adc_config.channelNumber = self->channel;
     adc_config.sampleScaleMode = kLPADC_SamplePartScale;
+    if (self->channel_group == 0) {
```

Comment:
Since kLPADC_SampleChannelSingleEndSideA has the value of 0 and kLPADC_SampleChannelSingleEndSideB the value of 1, you could directly assign `self->channel_group` (or whatever it will be called) to `adc_config.sampleChannelMode`. As long of course as the implementation supports single ended mode only.